lib/web/fetch/util.js

```diff
@@ -47,7 +47,7 @@ function responseLocationURL (response, requestFragment) {
     // Some websites respond location header in binary form without encoding them
     // and major browsers redirect them to correctly UTF-8 encoded addresses.
     // Here, we handle that behavior in the same way.
-    location = new URL(decodeBinaryStringToUtf8(location), responseURL(response))
+    location = new URL(normalizeBinaryStringToUtf8(location), responseURL(response))
```

Comment:
@lemire Would this be better? efccdde

Reply:
The benefit of Buffer.isAscii is that you calling on C++ which might be faster, but your code is nice too. I think it clarifies the issue as well.

Reply:
@lemire 
Could `Buffer.from` potentially incur significant costs? Since `buffer.isAscii` [does not accept a string input](https://nodejs.org/docs/latest-v20.x/api/buffer.html#bufferisasciiinput), I wrote it this way.

Reply:
@Xvezda What you did is good. I am not arguing. It is probably even better than my proposal because you sidestep entirely the creation of a Buffer.

Reply:
In my view, the new code is easier to understand. It is absolutely clear now that it has to do with non-ASCII content.

Reply:
I was just curious if the approach using Buffer.from might have more advantages. This should be sufficient then, thank you.

Reply:
Yes. I definitively think your code is fine.